src/lib/viz/style/helpers/color-bins-style.ts

```diff
@@ -62,17 +63,57 @@ export function colorBinsStyle(
 
     validateParameters(opts);
 
-    if (!opts.breaks.length) {
-      const stats = meta.stats.find(f => f.name === featureProperty) as NumericFieldStats;
-      const classifier = new Classifier(stats);
-      const breaks = classifier.breaks(opts.bins - 1, opts.method);
-      return calculateWithBreaks(featureProperty, breaks, meta.geometryType, opts);
+    return calculateWithBreaks(
+      featureProperty,
+      getBreaks(opts, meta, featureProperty),
+      meta.geometryType,
+      opts
+    );
+  };
+
+  const evalFNLegend = (layer: StyledLayer, properties = {}): LegendProperties[] => {
+    const meta = layer.source.getMetadata();
+
+    if (!meta.geometryType) {
+      return [];
     }
 
-    return calculateWithBreaks(featureProperty, opts.breaks, meta.geometryType, opts);
+    const opts = defaultOptions(meta.geometryType, options);
+    const breaks = getBreaks(opts, meta, featureProperty);
+    const stats = meta.stats.find(f => f.name === featureProperty) as NumericFieldStats;
+    let ranges = [...breaks, stats.max];
+    const colors = getColors(opts.palette, ranges.length);
+    ranges = [stats.min, ...ranges];
+    const styles = getStyles(meta.geometryType, opts) as any;
+    const geometryType = meta.geometryType.toLocaleLowerCase() as LegendGeometryType;
+
+    return colors.map((c, i) => {
+      return {
+        type: geometryType,
+        color: c,
+        label: `${ranges[i]} - ${ranges[i + 1]}`,
+        width: styles.getSize,
+        strokeColor:
+          geometryType !== 'line' && options.property !== 'strokeColor'
+            ? `rgba(${styles.getLineColor.join(',')})`
+            : undefined,
+        ...properties
+      };
+    });
   };
 
-  return new Style(evalFN, featureProperty);
+  return new Style(evalFN, featureProperty, evalFNLegend);
+}
+
+function getBreaks(opts: ColorBinsOptionsStyle, meta: SourceMetadata, featureProperty: string) {
```

Comment:
A question related to this. I see you have extracted the `getBreaks`function from the original place, which is nice. But I also see a small change regarding to how the number of breaks is passed. Is it keeping the same behaviour (part of the refactor) or is related with a detected bug on this part?

Reply:
If I have not been wrong, it is keeping the same behavior. In the original code in the line 69 you can see `return calculateWithBreaks(featureProperty, breaks, meta.geometryType, opts);` and in the line 72 `return calculateWithBreaks(featureProperty, opts.breaks, meta.geometryType, opts);` The only difference between both calls is the `breaks` or `opts.breaks` parameter. In the PR getBreak function returns `breaks` or `opts.breaks`